### PR TITLE
[PM-3797] Client changes to use new key rotation process

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Final Client changes for Key Rotation Improvements. 

- Introduces a new `KeyRotationService` that is responsible for owning rotation process.
- Moves `Send` re-encryption to the `SendService` (`KeyRotationService` shouldn't have knowledge about how domains are encrypted).
- Moves `EmergencyAccess` re-encryption to the `EmergencyAccessService`.
- Renames `AccountRecoveryService` to `OrganizationUserResetPasswordService` after feedback from Admin Console

## Code changes

Auth
- **emergency-access-update.request.ts:** New request model for domain updates that includes Id
- **emergency-access.service.ts:** Moved `EmergencyAccess` re-encryption to the `EmergencyAccessService`. Add deprecated method for legacy key rotations if feature flag is off
- **key-rotation.service/api/spec/module:** New key rotation service for owning the rotation process. Added api service, module, and spec file.
- **update-key.request.ts:** Moved to Auth ownership. Also added new properties for including other domains.
- **migrate-legacy-encryption.component.ts:** Use new key rotation service instead of old component specific service. Delete old service.
- **change-password.component.ts:** Use new key rotation service.
- **settings.module.ts:** Import key rotation module.

Admin Console
- **organization-user-reset-password.service.ts/spec:** Responsible for re-encryption of reset password keys during key rotation. Added tests.
- **organization-user-reset-password-enrollment.request.ts:** New request model for key rotations
- **reset-password.component.ts:** Update `AccountRecoveryService` to `OrganizationUserResetPasswordService`
- **enroll-master-password-reset.component.ts:** Update `AccountRecoveryService` to `OrganizationUserResetPasswordService`

Tools
- **send.service/spec.ts:** Responsible only for re-encryption of sends during key rotation. Added tests.

Other
- **api.service.ts:** Move `postAccountKey` to `KeyRotationApiService`
- **feature-flag.enum.ts:** add new feature flag

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Comment by jlf0dev** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6667** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/6667?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6881** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/6881?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/bitwarden/clients/6881?utm_source=stack-comment).

**Review by ike-kottlowski — APPROVED** [review]
A mighty improvement. Excellent work.

File: libs/common/src/tools/send/services/send.service.spec.ts [hunk 1]
```diff
@@ -92,6 +93,35 @@ describe("SendService", () => {
...
+      sendService.replace(null);
+
+      const newUserKey = new SymmetricCryptoKey(new Uint8Array(32)) as UserKey;
+      const result = await sendService.getRotatedKeys(newUserKey);
+
+      expect(result).toEqual([]);
+    });
+
+    it("throws if the new user key is null", async () => {
+      await expect(sendService.getRotatedKeys(null)).rejects.toThrow();
```
Comment (audrey-jensen) [inline libs/common/src/tools/send/services/send.service.spec.ts:121]: ⛏️ The throw should check to ensure that the message is correct.

**Review by audrey-jensen — COMMENTED** [review]
Just one nit ⛏️ . Thank you for documenting the new function! ❤️

File: libs/common/src/tools/send/services/send.service.spec.ts (hunk 1 above)
Comment (jlf0dev) [reply libs/common/src/tools/send/services/send.service.spec.ts]: Added!

File: libs/common/src/tools/send/services/send.service.ts [hunk 2]
```diff
@@ -212,6 +213,20 @@ export class SendService implements InternalSendServiceAbstraction {
     await this.stateService.setEncryptedSends(sends);
   }
 
+  async getRotatedKeys(newUserKey: UserKey): Promise<SendWithIdRequest[]> {
+    if (newUserKey == null) {
+      throw new Error("New user key is required for rotation.");
+    }
+
+    return await Promise.all(
```
[the remark this answers is not in the document]
Comment (jlf0dev) [reply libs/common/src/tools/send/services/send.service.ts]: Good point, I've definitely had frustrating debugging sessions dealing with the same thing. Updated!

**Review by addisonbeck — APPROVED** [review]

**Review by ike-kottlowski — APPROVED** [review]

**Pull request merged**